onnxruntime/core/platform/posix/env.cc

```diff
@@ -242,6 +262,37 @@ class PosixEnv : public Env {
     return common::Status(common::SYSTEM, e, oss.str());
   }
 
+  bool FolderExists(const std::string& path) const override {
+    struct stat sb;
+    if (stat(path.c_str(), &sb)) {
+      return false;
+    }
+    return S_ISDIR(sb.st_mode);
+  }
+
+  common::Status CreateFolder(const std::string& path) const override {
+    size_t pos = 0;
+    do {
+      pos = path.find_first_of("\\/", pos + 1);
```

Comment:
Prefer to use [https://linux.die.net/man/3/dirname](https://linux.die.net/man/3/dirname) , which is safer.

Reply:
Can you clarify how dirname would be safer? Here we are doing the equivalent of mkdir -p. dirname only gets the last directory, but we want any intermediate, nonexistent ones too.

---
In reply to: [392565039](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r392565039) [](ancestors = 392565039)

Reply:
@snnn I still don't see a clear advantage to using dirname. If I'm missing something let me know. Otherwise, we can refactor it in a future PR if desired.

---
In reply to: [393919604](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r393919604) [](ancestors = 393919604,392565039)